documentation/course/e2/README.md

```diff
@@ -2069,15 +2074,55 @@ TODO @jakubzehner: Dodać diagram pakietów, opis architektury i endpointy.
+| `admin`                                     | `PATCH`    | `/ext/v1/stops/:id`       | `LOG/09`                               | Edytowanie przystanku.                         |
+| `admin`                                     | `DELETE`   | `/ext/v1/stops/:id`       | `LOG/10`                               | Dezaktywacja przystanku.                       |
+| `passenger`, `admin`                        | `GET`      | `/ext/v1/routes`          | `LOG/11`, `LOG/12`                     | Pobranie rozkładu jazdy.                       |
+| `passenger`, `admin`                        | `GET`      | `/ext/v1/routes/line/:id` | `LOG/11`, `LOG/12`                     | Pobranie rozkładu jazdy dla linii.             |
+| `passenger`, `admin`                        | `GET`      | `/ext/v1/routes/:id`      | `LOG/11`, `LOG/12`                     | Pobranie informacji o kursie.                  |
+| `admin`                                     | `POST`     | `/ext/v1/routes`          | `LOG/13`,                              | Utworzenie nowego kursu.                       |
+| `admin`                                     | `PATCH`    | `/ext/v1/routes/:id`      | `LOG/14`                               | Edytowanie kursu.                              |
+| `admin`                                     | `DELETE`   | `/ext/v1/routes/:id`      | `LOG/15`                               | Deaktywacja kursu.                             |
+| `admin`                                     | `GET`      | `/ext/v1/vehicles`        | `LOG/16`                               | Pobranie listy pojazdów.                       |
+| `admin`                                     | `POST`     | `/ext/v1/vehicles`        | `LOG/17`                               | Utworzenie nowego pojazdu.                     |
+| `admin`                                     | `PATCH`    | `/ext/v1/vehicles/:id`    | `LOG/18`                               | Edytowanie pojazdu.                            |
+| `admin`                                     | `DELETE`   | `/ext/v1/vehicles/:id`    | `LOG/19`                               | Deaktywacja pojazdu.                           |
+| `passenger`, `driver`, `inspector`, `admin` | `GET`      | `/ext/v1/accidents`       | `LOG/20`, `LOG/21`, `LOG/24` `LOG/27`, | Pobranie listy awarii.                         |
+| `passenger`, `driver`, `inspector`, `admin` | `GET`      | `/ext/v1/accidents/:id`   | `LOG/20`, `LOG/21`, `LOG/24` `LOG/27`, | Pobranie informacji o awarii.                  |
+| `driver`, `inspector`                       | `POST`     | `/ext/v1/accidents`       | `LOG/22`, `LOG/25`                     | Zgłoszenie nowej awarii.                       |
+| `driver`, `inspector`                       | `PATCH`    | `/ext/v1/accidents/:id`   | `LOG/23`, `LOG/26`                     | Edytowanie awarii.                             |
+
```

Comment:
**suggestions:** You might want to consolidate some of the endpoints while also reducing the nesting. As an example:

- `/ext/v1/routes/line/:id` (or as in my suggestion `/ext/v1/lines/:id/routes`)
- `/ext/v1/stops/:id/routes` (unlisted, suggested by me)
- `/ext/v1/routes`

Could all be consolidated into: `/ext/v1/routes?stop=:id&line=:id`

In this case you:
- Have a smaller API area (which is good, because it means less breaking changes to make, and less OpenAPI specs to write).
- Have to write less code.
- Get the ability to filter by multiple variables for free (checking routes of a given line from a given stop).

Reply:
It looks like all complex `GET` requests can be replaced with additional query parameter to a simpler `GET` call.